test/test_pipeline.cpp

```diff
@@ -396,6 +398,29 @@ class automaticReshardingTest
       GTEST_EXPECT_TRUE(!isResharding(expr) || isLowerableToCommunication(expr))
           << "on expr=" << expr;
     }
+
+    SegmentCandidateFinderOptions options{
+        .run_translate_welford = false,
+        .run_combine_reductions = false,
+        .run_herrmann_merge = true,
+        .run_final_merge = true,
+        .only_segment_resharding_exprs = true};
+
+    KernelArgumentHolder dummy_inputs;
+    auto segmented_fusion = SegmentCandidateFinder::segment(
+        std::move(fusion), dummy_inputs, options);
+
+    for (auto group : segmented_fusion->groups()) {
```

Comment:
no problem, I make the change as you suggest. But for my understanding, why do we want to avoid `auto` here?

Reply:
Sorry, I should have said more about this. 

As a newcomer to the team, I had quite a lot troubles understanding what the `auto`s refer to in our codebase. In fact, I still have troubles today, just not as many 😄. 

Personally, I like [this guide about `auto`](https://google.github.io/styleguide/cppguide.html#Type_deduction) quite a lot explaining when and why to prefer explicit types. It makes code easier to read, and that's in general more important than easier to write (because code is read more often than written).

Reply:
Alright thanks !